alfresco-docker-transform-misc/src/main/java/org/alfresco/transformer/MiscController.java

```diff
@@ -109,24 +108,27 @@ public void processTransform(final File sourceFile, final File targetFile,
                 " '{}', timeout {} ms", sourceFile, targetFile, transformOptions, timeout);
         }
 
-        transformer.transform(sourceFile, targetFile, sourceMimetype, targetMimetype,
-            transformOptions);
+        String transform = getTransformerName(sourceFile, sourceMimetype, targetMimetype, transformOptions);
+        transformer.transform(transform, sourceFile, targetFile, sourceMimetype, targetMimetype, transformOptions);
     }
 
     @PostMapping(value = "/transform", consumes = MULTIPART_FORM_DATA_VALUE)
     public ResponseEntity<Resource> transform(HttpServletRequest request,
         @RequestParam("file") MultipartFile sourceMultipartFile,
         @RequestParam("targetExtension") String targetExtension,
         @RequestParam("targetMimetype") String targetMimetype,
+        @RequestParam(value = "targetEncoding", required = false) String targetEncoding,
         @RequestParam("sourceMimetype") String sourceMimetype,
-        @RequestParam(value = "testDelay", required = false) Long testDelay,
-        @RequestParam Map<String, String> parameters)
+        @RequestParam(value = "sourceEncoding", required = false) String sourceEncoding,

```

Comment:
I would prefer to keep the parameters like pageLimit out of the MiscController and only specify parameters common for all transformers, and if the creation of the transformOptions could be done in a more generic way so that addition of other transformers to the Misc T-engine will not require unnecessary changes to other classes.